scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -4010,39 +4191,48 @@ def gather_sample(self):
 
         # gather metrics
         containers = None
-        if self.__report_container_metrics and self.__client:
-            if (
-                self.__metrics_controlled_warmer is not None
-                and not self.__metrics_controlled_warmer.is_running()
-            ):
-                self.__metrics_controlled_warmer.set_k8s_cache(k8s_cache)
-                self.__metrics_controlled_warmer.start()
-            containers = _get_containers(
-                self.__client,
-                ignore_container=None,
-                glob_list=self.__glob_list,
-                k8s_cache=k8s_cache,
-                k8s_include_by_default=self.__include_all,
-                k8s_namespaces_to_include=self.__namespaces_to_include,
-                controlled_warmer=self.__metrics_controlled_warmer,
-            )
+        if self.__report_container_metrics:
+            if self.__client:
+                if (
+                    self.__metrics_controlled_warmer is not None
+                    and not self.__metrics_controlled_warmer.is_running()
+                ):
+                    self.__metrics_controlled_warmer.set_k8s_cache(k8s_cache)
+                    self.__metrics_controlled_warmer.start()
+                containers = _get_containers(
+                    self.__client,
+                    ignore_container=None,
+                    glob_list=self.__glob_list,
+                    k8s_cache=k8s_cache,
+                    k8s_include_by_default=self.__include_all,
+                    k8s_namespaces_to_include=self.__namespaces_to_include,
+                    controlled_warmer=self.__metrics_controlled_warmer,
+                )
+            else:
+                containers = self.__container_checker.get_containers()
         try:
             if containers:
+                stats = None
                 if self.__report_container_metrics:
                     self._logger.log(
                         scalyr_logging.DEBUG_LEVEL_3,
                         "Attempting to retrieve metrics for %d containers"
                         % len(containers),
                     )
-                    self.__gather_metrics_from_api(containers, cluster_name)
+                    if self.__metric_fetcher:
```

Comment:
Probably also a good idea to log some rate limited info message here (e.g. once per day) something along the lines of - retrieving container metrics from Docker socket / kubeletes API.